pkg/core/types.go

```diff
@@ -198,6 +199,19 @@ type Provider interface {
 	DeleteMapping(p KeyPath) error
 }
 
+type MetaInfo struct {
+	Description    string
+	Authentication string
+	ConfigTemplate string
+	Ops            OpMatrix
+}
+type OpMatrix struct {
+	Sync    bool
+	Mapping bool
+	Read    bool
+	Write   bool
```

Comment:
We have providers (Consul, etcd for example) that implement only put -> "write" without delete. 
I think we have two options:
1. Split the `Write` to `Put` and `Delete.`
2. `Write` implementation required both of the action types.

WDTY?

Reply:
Yea, splitting sounds good